code/game/objects/items/shields.dm

```diff
@@ -104,7 +104,7 @@
 	transparent = TRUE
 
 /obj/item/shield/riot/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/melee/baton))
+	if(istype(W, /obj/item/melee/baton) || istype(W, /obj/item/melee/tonfa))
```

Comment:
should probably include /obj/item/melee/classic_baton subtypes too, no? 

It should clunk if I hit it with a tonfa, stunbaton, Billy club, or det baton

Reply:
I'll let it work for any melee weapon that isn't sharp